community/modules/scripts/ramble-execute/main.tf

```diff
@@ -15,34 +15,69 @@
  */
 
 locals {
-  commands_content = var.commands == null ? "" : indent(4, yamlencode(var.commands))
+  # This label allows for billing report tracking based on module.
+  labels = merge(var.labels, { ghpc_module = "ramble-execute", ghpc_role = "scripts" })
+}
+
+locals {
+  commands_content = var.commands == null ? "echo 'no ramble commands provided'" : indent(4, yamlencode(var.commands))
 
   execute_contents = templatefile(
     "${path.module}/templates/ramble_execute.yml.tpl",
     {
-      pre_script = ". ${var.spack_path}/share/spack/setup-env.sh && . ${var.ramble_path}/share/ramble/setup-env.sh"
+      pre_script = ". /etc/profile.d/spack.sh && . /etc/profile.d/ramble.sh"
```

Comment:
IIUC you would like the option to use ramble-execute without spack-setup or even without ramble-setup. Previously the existence of the [spack|ramble]_path variables required that these were present. 

This pre script seems problematic since we now can't be sure that either of these files exist. Should we be testing for the files before calling them?

Reply:
Not exactly.

I would like to be able to call `ramble-setup` and `ramble-execute` as separate runners, rather than needing them to be chained.

So, this new format just requires that `*-setup` has run successfully on the machine you're trying to run `*-execute` on. Since both of them create the `/etc/profile.d/*.sh` scripts.

I updated it to be conditional though. Let me know if you dislike the change.

Reply:
discussed and agreed to offline.